File: src/de/schauderhaft/cleanCode/exercise6/LabelGenerator.java
package de.schauderhaft.cleanCode.exercise6;

import javax.swing.JLabel;

public class LabelGenerator {

    JLabel stageInfoLabel;

    TextStrategy textGenerator = new TextStrategy();
    VisabilityStrategy visabilityStrategy = new VisabilityStrategy();

    public JLabel getLblStageInfo() {
        if (stageInfoLabel == null) {
            stageInfoLabel = new JLabel();

            stageInfoLabel.setText(textGenerator.getText());
            stageInfoLabel.setVisible(visabilityStrategy.getVisibility());
        }

        return stageInfoLabel;
    }

}

File: src/de/schauderhaft/cleanCode/exercise6/LabelGenerator.cs
using System.Windows.Forms;

namespace de.schauderhaft.cleanCode.exercise6
{
    public class LabelGenerator
    {
        private Label stageInfoLabel;
        private TextStrategy textGenerator = new TextStrategy();
        private VisibilityStrategy visibilityStrategy = new VisibilityStrategy();
        
        public Label GetLblStageInfo()
        {
            if (stageInfoLabel == null)
            {
                stageInfoLabel = new Label();
                
                stageInfoLabel.Text = textGenerator.GetText();
                stageInfoLabel.Visible = visibilityStrategy.GetVisibility();
            }
            
            return stageInfoLabel;
        }
    }
}